src/components/Navbar/index.tsx

```diff
@@ -1,40 +1,69 @@
-import React, { ReactElement } from "react"
-import { ReactComponent as ProfileIcon } from "../../assets/profile.svg"
+import React from "react"
+import { gql, useQuery } from "@apollo/client"
 import { ReactComponent as StarIcon } from "../../assets/star.svg"
-interface NavbarProps {
-  Logo: ReactElement
-}
-export default function Navbar({ Logo }: NavbarProps) {
+import { FilterCharacter, INavbarProps } from "../../types/interfaces"
+import SearchBar from "./_searchBar"
+
+export default function Navbar({ Logo }: INavbarProps) {
   const [searchValue, setSearchValue] = React.useState("")
 
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(e.target.value)
+    refetch({
+      filter: {
+        name: e.target.value,
+      },
+    })
```

Comment:
It would be good to have a debounce search here. Otherwise it will search everytime you hit a button.